examples/experimental/omniglot_protonet_pytorch/data.py

```diff
@@ -0,0 +1,113 @@
+import os
+import numpy as np
+
+import torch
+from torch.utils.data import Dataset
+
+from PIL import Image
+
+# Each training example will be a classification task with n classes.
+
+
+class OmniglotTasks(Dataset):
+    def __init__(
+        self,
+        data_path,
+        tasks_per_epoch,
+        class_inds,
+        img_resize_dim,
+        num_classes,
+        num_support,
+        num_query=None,
+        rotations=(0, 90, 180, 280),
+    ):
+        """
+        Each subfolder of the parent data directory is a separate
+        alphabet with multiple characters.  Each class is a certain
+        character within an alphabet.  Tasks will correspond to 
+        n-way classification problems where the learner will 
+        have to predict which of n classes a test image belongs to.
+
+        Args:
+            # NOTE: You should partition all inds to a train set and a validation
+            # set of indices and use those as args here.
+            class_inds: character inds to sample from to generate tasks.
+            num_support: how many image per class to use to construct class prototypes
+            num_query: how many images per class to use to update embedding
+
+        """
+        self.class_inds = class_inds
+        self.rotations = rotations
+        self.tasks_per_epoch = tasks_per_epoch
+        self.img_resize_dim = img_resize_dim
+
+        self.class_paths = {}
+        class_idx = 0
+        min_class_examples = float("inf")
+        for root, dirs, files in os.walk(data_path):
+            if len(dirs) == 0:
+                min_class_examples = min(min_class_examples, len(files))
+                self.class_paths[class_idx] = [os.path.join(root, f) for f in files]
+                class_idx += 1
+
+        self.num_classes = num_classes
+        self.num_support = num_support
+        self.num_query = (
+            min_class_examples - num_support if (num_query is None) else num_query
+        )
+
+    def get_collate_fn(self):
```

Comment:
little confused here. Is this just using the default collate function? If so, we shouldn't need to define it.

Reply:
this is different from the default collate_fn.  default zips dictionary field values and returns a list for a given field whereas I am creating a list of dictionaries.

Reply:
Let's add some comments then describing this.

Reply:
added comment